osc.c

```diff
@@ -1637,13 +1637,16 @@ static gboolean idle_timeout_check(gpointer ptr)
 {
 	int ret;
 	struct iio_context *new_ctx = (struct iio_context *) ptr;
+	/* Just get the first dev. We only run this if there are devices in the IIO context */
+	struct iio_device *dev = iio_context_get_device(new_ctx, 0);
```

Comment:
Wouldn't it make sense to get device after this check: if (new_ctx != ctx)? Just in case ptr is NULL or even if new_ctx is just different from ctx.

Reply:
I think in the end we'll be fine like this :). But I will change to your suggestion